src/devices/Ft4222/Ft4222Spi.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Ft4222;
+using System.IO;
+
+namespace System.Device.Spi
+{
+    /// <summary>
+    /// Create a SPI Device based on FT4222 chipset
+    /// </summary>
+    public class Ft4222Spi : SpiDevice
+    {
+        private readonly SpiConnectionSettings _settings;
+        private IntPtr _ftHandle = new IntPtr();
+
+        /// <summary>
+        /// The connection settings of a device on a SPI bus. The connection settings are immutable after the device is created
+        /// so the object returned will be a clone of the settings object.
+        /// </summary>
+        public override SpiConnectionSettings ConnectionSettings => _settings;
+
+        /// <summary>
+        /// Store the FTDI Device Information
+        /// </summary>
+        public DeviceInformation DeviceInformation { get; internal set; }
+
+        /// <summary>
+        /// Create an SPI FT4222 class
+        /// </summary>
+        /// <param name="settings">SPI Connection Settings</param>
+        public Ft4222Spi(SpiConnectionSettings settings)
+        {
+            _settings = settings;
+            // Check device
+            var devInfos = FtCommon.GetDevices();
+            // Select the one from bus Id
+            DeviceInformation = devInfos[_settings.BusId];
+
+            // Open device
+            var ftStatus = FtFunction.FT_OpenEx(DeviceInformation.LocId, FtOpenType.OpenByLocation, ref _ftHandle);
+
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to open device {DeviceInformation.Description}with error: {ftStatus}");
+
+            // Set the clock but we need some math
+            var (ft4222Clock, tfSpiDiv) = CalculateBestClockRate();
+
+            ftStatus = FtFunction.FT4222_SetClock(_ftHandle, ft4222Clock);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed set clock rate {ft4222Clock} on device: {DeviceInformation.Description}with error: {ftStatus}");
+
+
+            SpiClockPolarity pol = SpiClockPolarity.ClockIdleLow;
```

Comment:
I called it completely differently in SoftwareSpi :smile: Presumably those are correct names for those settings